### PR TITLE
Atomicity check

### LINKED ISSUE #282: Atomicity check test
There is no atomicity check test in the test suit. We should have one.

### PR DESCRIPTION
**Pull request by Sammers21**
Closes #282

### CODE REVIEW

File: src/test/java/com/artipie/asto/StorageSaveAndLoadTest.java [hunk 1]
```diff
@@ -80,6 +83,39 @@ void shouldSaveFromMultipleBuffers(final Storage storage) throws Exception {
...
+                key,
+                new Content.OneTime(
+                    new Content.From(
+                        Flowable.concat(
+                            Flowable.just(ByteBuffer.wrap(new byte[]{1})),
+                            Flowable.error(new IllegalStateException())
+                        )
+                    )
+                )
+            ).get();
```
Comment (olegmoz) [inline src/test/java/com/artipie/asto/StorageSaveAndLoadTest.java:109]: @Sammers21 we can use `.join()` here and avoid `throws` in method signature

File: src/test/java/com/artipie/asto/StorageSaveAndLoadTest.java [hunk 2]
```diff
@@ -80,6 +83,39 @@ void shouldSaveFromMultipleBuffers(final Storage storage) throws Exception {
         );
     }
 
+    @SuppressWarnings("PMD.EmptyCatchBlock")
+    @TestTemplate
+    @Timeout(1)
+    void saveIsAtomic(final Storage storage) throws ExecutionException, InterruptedException {
```
Comment (olegmoz) [inline src/test/java/com/artipie/asto/StorageSaveAndLoadTest.java:89]: @Sammers21 this test is very similar to `shouldNotSavePartial` test we already have. The only difference is that we check situation with previously existing value. I'd suggest to rename this method name `shouldNotOverwriteWithPartial` so similar tests will have similar names

**Review by olegmoz — CHANGES_REQUESTED** [review]
@Sammers21 thanks, please see comments

**Comment by Sammers21** [comment]
@olegmoz, review the PR one more time, please.

**Review by olegmoz — APPROVED** [review]
@Sammers21 thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/asto#283` is not assigned, can't get performer

**Pull request merged**